providers/ldap/ad/ad_provider.go

```diff
@@ -0,0 +1,281 @@
+	UserEnabledAttributeSetting     = settingBase + "user.enabled.attribute"
+	GroupNameFieldSetting           = settingBase + "group.name.field"
+	TlsSetting                      = settingBase + "tls"
+	TimeoutSetting                  = settingBase + "connection.timeout"
+	AllowedIdentitiesSetting        = settingBase + "allowed.identities"
+	GroupDnFieldSetting             = settingBase + "group.dn.field"
+	GroupMemberUserAttributeSetting = settingBase + "group.member.user.attribute"
+)
+
+var scopes = []string{UserScope, GroupScope}
+
+type ADProvider struct {
+	LdapClient *ldap.LClient
+}
+
+var adConstantsConfig = &ldap.ConstantsConfig{
+	UserScope:            UserScope,
+	GroupScope:           GroupScope,
+	Scopes:               scopes,
+	MemberOfAttribute:    MemberOfAttribute,
+	ObjectClassAttribute: ObjectClassAttribute,
+	LdapJwt:              LdapJwt,
+}
+
+func InitializeProvider() *ADProvider {
+	ldapClient := &ldap.LClient{}
+	ldapProvider := &ADProvider{}
+
+	ldapClient.ConstantsConfig = adConstantsConfig
+	ldapProvider.LdapClient = ldapClient
+
+	return ldapProvider
+}
+
+func (a *ADProvider) GetName() string {
+	return Name
+}
+
+func (a *ADProvider) GetUserType() string {
+	return UserScope
+}
+
+func (a *ADProvider) GenerateToken(jsonInput map[string]string) (model.Token, error) {
+	return a.LdapClient.GenerateToken(jsonInput)
+}
+
+func (a *ADProvider) GetIdentity(distinguishedName string, scope string, accessToken string) (client.Identity, error) {
+	return a.LdapClient.GetIdentity(distinguishedName, scope)
+}
+
+//SearchIdentities returns the identity by name
+func (a *ADProvider) SearchIdentities(name string, exactMatch bool, accessToken string) ([]client.Identity, error) {
+	return a.LdapClient.SearchIdentities(name, exactMatch)
+}
+
+//GetSettings transforms the provider config to db settings
+func (a *ADProvider) GetSettings() map[string]string {
+	settings := make(map[string]string)
+
+	settings[DomainSetting] = a.LdapClient.Config.Domain
+	settings[GroupSearchDomainSetting] = a.LdapClient.Config.GroupSearchDomain
+	settings[LoginDomainSetting] = a.LdapClient.Config.LoginDomain
+	settings[PortSetting] = strconv.FormatInt(a.LdapClient.Config.Port, 10)
+	settings[UserSearchFieldSetting] = a.LdapClient.Config.UserSearchField
+	settings[ServiceAccountUsernameSetting] = a.LdapClient.Config.ServiceAccountUsername
+	settings[GroupSearchFieldSetting] = a.LdapClient.Config.GroupSearchField
+	settings[UserObjectClassSetting] = a.LdapClient.Config.UserObjectClass
+	settings[UserNameFieldSetting] = a.LdapClient.Config.UserNameField
+	settings[GroupObjectClassSetting] = a.LdapClient.Config.GroupObjectClass
+	settings[UserLoginFieldSetting] = a.LdapClient.Config.UserLoginField
+	settings[UserDisabledBitMaskSetting] = strconv.FormatInt(a.LdapClient.Config.UserDisabledBitMask, 10)
+	settings[ServerSetting] = a.LdapClient.Config.Server
+	settings[ServiceAccountPasswordSetting] = a.LdapClient.Config.ServiceAccountPassword
```

Comment:
Set the ServiceAccountPasswordSetting in settings[] only if a.LdapClient.Config.ServiceAccountPassword != ""
Check github/shib. This is done because UI will not send the password everytime for security